components/diagnose/scan.go

```diff
@@ -214,6 +215,31 @@ func Scan(ctx context.Context, opts ...OpOption) error {
 					}
 				}
 			}
+
+			if op.checkIb {
+				cctx, cancel := context.WithTimeout(ctx, 15*time.Second)
```

Comment:
The function is too long.. could you wrap these codes as a function?

Reply:
Are these codes duplicated in `accelerator/nvidia/infiniband`? we're going to directly call `ibstat` to detect if there is any ib issue. I think these codes are similar, but different rate configuration..

Reply:
Yes, will be similar, the difference is that, the `accelerator/nvidia/infiniband` component code will set the state to unhealthy

Reply:
Yeah, so it would be better to implement it somewhere that can be reused by `accelerator/nvidia/infiniband` or just put it in `accelerator/nvidia/infiniband` if possible.

Reply:
done